lightning/src/ln/peer_handler.rs

```diff
@@ -765,11 +790,10 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref, CMH: Deref> P
 					Some(buff) => buff,
 				};
 
-				let should_be_reading = peer.pending_outbound_buffer.len() < OUTBOUND_BUFFER_LIMIT_READ_PAUSE;
 				let pending = &next_buff[peer.pending_outbound_buffer_first_msg_offset..];
-				let data_sent = descriptor.send_data(pending, should_be_reading);
+				let data_sent = descriptor.send_data(pending, peer.should_read());
 				peer.pending_outbound_buffer_first_msg_offset += data_sent;
-				if peer.pending_outbound_buffer_first_msg_offset == next_buff.len() { true } else { false }
+				peer.pending_outbound_buffer_first_msg_offset == next_buff.len()
 			} {
 				peer.pending_outbound_buffer_first_msg_offset = 0;
 				peer.pending_outbound_buffer.pop_front();
```

Comment:
Could this entire `if-else` block be moved into a `send_data` method taking a `Descriptor` on `Peer`? It would be better encapsulated that way, I'd imagine.

Reply:
Yeah, removing the `if {}` pattern would be preferred. That was initially going to be my suggestion before going a step further.